develop/tutorials/articles/100-tooling/04-liferay-workspace/06-validating-modules-against-the-target-platform.markdown

```diff
@@ -0,0 +1,236 @@
+
+1.  Start the @product@ instance stored in your workspace. Make sure the
+    platform you want to depend on is installed.
+
+2.  Download the
+    [BND Remote Agent JAR file](https://search.maven.org/#search%7Cga%7C1%7Cbiz.aqute.remote.agent)
+    and copy it into your @product@ instance's `osgi/modules` folder.
+
+3.  From the root folder of you workspace, run the following command:
+
+        bnd remote distro -o custom_distro.jar com.liferay.ce.portal.distro 7.1.0-m1
+
+    <!-- TODO: Update milestone version with correct syntax, when available. -Cody -->
+
+    This connects to the newly deployed BND agent running in @product@ and
+    generates a new distro JAR named `custom_distro.jar`. All other capabilities
+    will be overridden based on the @product@ instance you're generating the
+    custom distro against too, so verify the workspace bundle is the version you
+    plan to release for in production.
+
+4.  Navigate to your workspace's root `build.gradle` file and add the following
+    dependency:
+
+        dependencies {
+            targetPlatformDistro files('custom_distro.jar')
+        }
+
+Now your workspace is pointing to a custom distro JAR file instead of the
+default one provided. Run the `resolve` task to validate your custom modules
+against the custom set of capabilities.
+
+## Validating Modules Outside of Workspace
+
+<!-- TODO: Remove this section once the reference doc for the Gradle plugin is
+available. This is just gives a taste of targeting a platform for those who
+don't want to use Workspace. More details will be provided in reference doc.
+-Cody -->
+
+If you prefer to not use Liferay Workspace, but still want to validate modules
+against a target platform, you must apply the
+[Target Platform](https://github.com/liferay/liferay-portal/tree/master/modules/sdk/gradle-plugins-target-platform)
+Gradle plugin to the root `build.gradle` file of your custom multi-module Gradle
+build. Follow the
+[Targeting a Platform Outside of Workspace](/develop/tutorials/-/knowledge_base/7-1/managing-the-target-platform-for-liferay-workspace#targeting-a-platform-outside-of-workspace)
+section to do this.
+
+Once you have the Target Platform plugin and its BOM dependencies configured,
+you must configure the `targetPlatformDistro` dependency. Open your project's
+root `build.gradle` file and add it to the list of dependencies. It should look
+like this:
+
+    dependencies {
+        targetPlatformBoms group: "com.liferay", name: "com.liferay.ce.portal.bom", version: "7.1.0-m1"
+        targetPlatformBoms group: "com.liferay", name: "com.liferay.ce.portal.compile.only", version: "7.1.0-m1"
+        targetPlatformDistro group: "com.liferay", name "com.liferay.ce.portal.distro", version: "7.1.0-m1"
+    }
+
+<!-- TODO: Update above versions to accurate milestone version syntax. This was
+a shot in the dark. -Cody -->
+
+Now you can validate your modules against a target platform!
```

Comment:
We should have a section for how to fix common resolution errors

1) missing import packages
2) missing service capabilities
3) missing basic requirements such as `JavaSE version=8`

Reply:
This section will be important for helping our users how to "fix" the types of errors that the resolve task find.

Reply:
Also we need a two new sections

1) how to ignore resolve errors for particular modules 
2) how to configure workspace in such a way that the `resolve` task will execute as a part of gradle's `check` task group.  basically in their build.gradle of the project where they want to add this they just say `check.dependsOn resolve`  This can be done at the top level or in individual sub-projects.  Note: there is no real reason to do both :)  doing it at the top level is as good as doing it individually.

Reply:
@gamerson you mentioned creating a new section for "how to ignore resolve errors for particular modules." Isn't this the scenerio using this code?
```targetPlatform {
        resolveOnlyIf { project ->
            project.name != 'PROJECT_NAME'
        }
    }
```
Or is this something different?